### PR TITLE
Release 166 0

### PR DESCRIPTION
**Pull request by bijayprasadtiwari**
Added CRUD operations for whitelist
- View 
- View Share

### CODE REVIEW

File: src/Cohesity.Powershell/Scripts/View/Set-CohesityViewWhitelist.ps1 [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        .DESCRIPTION
+        Add whitelist IP(s) for a given view.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Set-CohesityViewWhitelist -ViewName view1 -ViewWhitelist $viewWhitelist
+        Get the whitelist as follows, $viewWhitelist = Get-CohesityViewWhitelist -ViewName view1
+        Set whitelist for a given view.
```
Comment (pyashish) [inline src/Cohesity.Powershell/Scripts/View/Set-CohesityViewWhitelist.ps1:14]: What's the use of this? Can someone change property of the view ? if yes, then can that be part of the example ?

File: src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1 [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        Add whitelist IP(s) for a given view.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Add-CohesityViewWhitelist -IP4List "1.1.1.1", "2.2.2.2" -NetmaskIP4 "255.255.255.0"
+        Add whitelist IP(s) an override global whitelist for a given view.
+        .EXAMPLE
+        Add-CohesityViewWhitelist -IP4List "1.1.1.1", "2.2.2.2" -NetmaskIP4 "255.255.255.0" -NFSRootSquash -NFSAccess "kReadWrite" -NFSAllSquash -SMBAccess "kReadWrite"
```
Comment (pyashish) [inline src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1:15]: Of late got another request: 

```
I see the Powershel Properties , Its taking the SID not the User ID
```

Can we solve setting ACL / whitelist with a single cmdlet or setter ?

File: src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1 [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    #>
+    [OutputType('System.Collections.ArrayList')]
+    [CmdletBinding(SupportsShouldProcess = $True, ConfirmImpact = "High")]
+    Param(
+        [Parameter(Mandatory = $true)]
+        # Specifies view name.
+        [string]$ViewName,
+        [Parameter(Mandatory = $true)]
+        # Specifies IPv4 addresses.
+        [string[]]$IP4List,
```
Comment (pyashish) [inline src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1:26]: Do we support ipv6 , if not, should we just put this as  IPAllowlist , also these could be FQDNs
?

File: src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1 [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Add-CohesityViewWhitelist -IP4List "1.1.1.1", "2.2.2.2" -NetmaskIP4 "255.255.255.0"
+        Add whitelist IP(s) an override global whitelist for a given view.
+        .EXAMPLE
+        Add-CohesityViewWhitelist -IP4List "1.1.1.1", "2.2.2.2" -NetmaskIP4 "255.255.255.0" -NFSRootSquash -NFSAccess "kReadWrite" -NFSAllSquash -SMBAccess "kReadWrite"
+        Add whitelist IP(s) an override global whitelist for a given view with optional parameters
```
Comment (pyashish) [inline src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1:16]: We have started a project to have inclusive names. Can you please change whitelist to AllowList everywhere ? 

If the cmdlet is already released. lets introduce a new param/cmdlet and mark the older one deprecated. Give it 6 months and then deprecate the older one.

**Review by pyashish — CHANGES_REQUESTED** [review]

File: src/Cohesity.Powershell/Scripts/View/Set-CohesityViewWhitelist.ps1 (hunk 1 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/View/Set-CohesityViewWhitelist.ps1]: Cannot change other properties of view, user will be allowed to change only the allowlist for the view.

File: src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1 (hunk 2 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1]: We can have a single setter for whitelist. Did not get the context for "Its taking the SID not the User ID".

File: src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1 (hunk 3 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1]: Done.

File: src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1 (hunk 4 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/View/Add-CohesityViewWhitelist.ps1]: We do not have any param/cmdlet with non-inclusive items. Will avoid using non-inclusive names for the future cmdlet/params.

**Comment by bijayprasadtiwari** [comment]
> Was this existing cmdlet or part of this new one ?

This file is for response display format. You can customize your display.

**Pull request merged**